File: app/schemas/rp.py
from pydantic import BaseModel, Field


class RPCreate(BaseModel):
    cdp_numero: int
    nit_tercero: str
    valor: float = Field(gt=0)
    objeto: str


class RPUpdate(BaseModel):
    valor: float | None = Field(default=None, gt=0)
    objeto: str | None = None
    fuente_sifse: int | None = None
    item_sifse: int | None = None


class RPResponse(BaseModel):
    numero: int
    fecha: str
    cdp_numero: int
    codigo_rubro: str
    cuenta: str | None = None
    nit_tercero: str
    nombre_tercero: str | None = None
    valor: float
    objeto: str
    estado: str
    fuente_sifse: int
    item_sifse: int
    saldo: float | None = None

    model_config = {"from_attributes": True}

File: frontend/src/lib/types/rp.ts
export interface RP {
  numero: number;
  fecha: string;
  cdp_numero: number;
  codigo_rubro: string;
  cuenta?: string | null;
  nit_tercero: string;
  nombre_tercero?: string | null;
  valor: number;
  objeto: string;
  estado: string;
  fuente_sifse: number;
  item_sifse: number;
  saldo?: number | null;
}

export interface RPCreate {
  cdp_numero: number;
  nit_tercero: string;
  valor: number;
  objeto: string;
}

export interface RPUpdate {
  valor?: number;
  objeto?: string;
  fuente_sifse?: number;
  item_sifse?: number;
}
